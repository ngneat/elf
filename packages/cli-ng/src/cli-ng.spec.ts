import { createRepo } from '@ngneat/elf-cli';
import ngHook from './index';

describe('cli-ng', () => {
  it('should add the Injectable decorator', () => {
    const output = createRepo({
      idKey: 'id',
      features: ['withProps'],
      crud: [],
      template: 'class',
      storeName: 'todos',
      hooks: [ngHook],
      path: '',
    });

    expect(output).toMatchSnapshot();
  });
});
