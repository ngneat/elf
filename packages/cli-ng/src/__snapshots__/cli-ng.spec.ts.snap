// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli-ng should add the Injectable decorator 1`] = `
"import { createStore, withProps } from '@ngneat/elf';
import { Injectable } from '@angular/core';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface TodosProps {
}

const store = createStore({ name: 'todos' }, withProps<TodosProps>({}));

@Injectable({ providedIn: 'root' })
export class TodosRepository {
}
"
`;
